### PR TITLE
👷 Export junit xml to datadog

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Datadog ci product does not support our tests frameworks but do support junit.xml files.
Since we already generates some, we could start to upload them to datadog.

## Changes

- Add export-test-result ci job
- Update gitlab-ci.yml

## Testing

Gitlab

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: scripts/export-test-result.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const { getSecretKey, executeCommand, printLog, logAndExit } = require('./utils')
+
+async function main() {
+  await exportTestResult('unit', ['test-report/unit/', 'test-report/unit-bs/'])
+  await exportTestResult('e2e', ['test-report/e2e/', 'test-report/e2e-bs/'])
+}
+
+async function exportTestResult(type, folders) {
+  const ddCiApiKey = (await getSecretKey('ci.browser-sdk.datadog_ci_api_key')).trim()
```
Comment (BenoitZugmeyer) [inline scripts/export-test-result.js:11]: Maybe `getSecretKey` could `trim` the key itself (for all secret keys)?

File: scripts/export-test-result.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  await executeCommand(
+    `npx @datadog/datadog-ci junit upload --service browser-sdk --env ci --tags type:${type} ${folders.join(' ')}`,
```
Comment (BenoitZugmeyer) [inline scripts/export-test-result.js:14]: It's a bit sad that it can't be used as a library. Anyway, I'd use `yarn` instead of `npx` since it is our package manager of choice :)

```suggestion
    `yarn datadog-ci junit upload --service browser-sdk --env ci --tags type:${type} ${folders.join(' ')}`,
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -225,8 +237,7 @@ deploy-prod-canary:
   stage: deploy:canary
   extends:
     - .base-configuration
-    - .tags
-  when: manual
```
Comment (BenoitZugmeyer) [inline .gitlab-ci.yml:229]: Some unrelated change here

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -286,7 +297,7 @@ notify-prod-canary-success:
   stage: notify:canary
   extends:
     - .prepare_notification
-    - .tags
```
Comment (BenoitZugmeyer) [inline .gitlab-ci.yml:289]: Here too

File: scripts/export-test-result.js (hunk 1 above)
Comment (amortemousque) [reply scripts/export-test-result.js]: I say, yes!
